src/screens/TokenSwap.js

```diff
@@ -0,0 +1,394 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { numberUtils } from '@hathor/wallet-lib';
+import React, { useState, useEffect } from 'react';
+import {
+  Keyboard,
+  KeyboardAvoidingView,
+  Pressable,
+  StyleSheet,
+  Text,
+  View,
+  TouchableOpacity,
+} from 'react-native';
+import { SwapIcon } from '../components/Icons/Swap.icon';
+import { useDispatch, useSelector } from 'react-redux';
+import { getStatusBarHeight } from 'react-native-status-bar-height';
+import { t, ngettext, msgid } from 'ttag';
+import { get } from 'lodash';
+
+import { renderValue } from '../utils';
+import NewHathorButton from '../components/NewHathorButton';
+import AmountTextInput from '../components/AmountTextInput';
+import InputLabel from '../components/InputLabel';
+import TokenBox from '../components/TokenBox';
+import HathorHeader from '../components/HathorHeader';
+import OfflineBar from '../components/OfflineBar';
+import { COLORS } from '../styles/themes';
+import { useNavigation } from '../hooks/navigation';
+import {
+  tokenSwapSetInputToken,
+  tokenSwapSetOutputToken,
+  tokenSwapSwitchTokens,
+} from '../actions';
+
+
+const SwapDivider = ({ onPress }) => {
+  return (
+    <View style={styles.dividerContainer}>
+      <View style={styles.dividerLine} />
+        <TouchableOpacity style={styles.dividerButton} onPress={onPress}>
+          <SwapIcon color={COLORS.white} />
+        </TouchableOpacity>
+      <View style={styles.dividerLine} />
+    </View>
+  );
+};
+
+const TokenSwap = () => {
+  const dispatch = useDispatch();
+  const tokensBalance = useSelector((state) => state.tokensBalance);
+
+  const [inputTokenAmountStr, setInputTokenAmountStr] = useState('0');
+  const [inputTokenAmount, setInputTokenAmount] = useState(0n);
+
+  const [outputTokenAmountStr, setOutputTokenAmountStr] = useState('0');
+  const [outputTokenAmount, setOutputTokenAmount] = useState(0n);
+
+  const [swapDirection, setSwapDirection] = useState(null);
+
+  // XXX: this is a mock value to simulate fetching and showing the quote section
+  const [showQuote, setShowQuote] = useState(false);
+
+  const {
+    allowedTokens,
+    inputToken,
+    outputToken,
+  } = useSelector((state) => state.tokenSwap);
+  const { decimalPlaces } = useSelector((state) => ({
+    decimalPlaces: state.serverInfo?.decimal_places
+  }));
+
+  const navigation = useNavigation();
+
+  useEffect(() => {
+    if (!inputToken) {
+      dispatch(tokenSwapSetInputToken(allowedTokens[0]));
```

Comment:
issue: For now this is not a problem as the api responses are mocked, but in the future could the `allowedTokens` be an empty array for some edge case reason?

If so, we should have at least a simple protection to avoid crashing the app here.

Reply:
This is an interesting observation, since this is a "swap" service it makes no sense to deploy a list with less than 2 tokens so I didn't even consider this possibility because having 1 token on the list makes the entire feature unavailable.

Maybe we should check for this length on the saga that requests the allowed tokens list and just put an error in this case, what do you think?

Reply:
I agree that's the best approach! Rejecting the received allowedTokens lists if it has less than two tokens in the saga.

What I'm trying to avoid here is a crash for something so small such as an array out of bounds.

Reply:
Done